### ISSUE TITLE
Amenity=place_of_worship buildings

### ISSUE THREAD

**Issue by pnorman** [gravitystorm/openstreetmap-carto#1206]
When I initially rewrote the buildings in #565, I wanted to take out the special handling of buildings that had an `amenity=place_of_worship` tag because it didn't work well with churches that weren't P.O.W.s, churches as nodes inside buildings, and anything other than the simplest `amenity=place_of_worship building=*`.

If we want places of worship to be rendered more important, we should change the POI rendering. If we want churches to be rendered more important, the current code does not do that.

I can write code that performs the spatial queries for checking for P.O.W. POIs inside a building, but we need to figure out if that's what we want.

Cross-ref #1176 #1205

**Comment by HolgerJeromin** [comment]
Some churches are tagged as buildings with a node of the PoW inside. At least in Germany this is in most cases a not optimal tagging. 
IMO if a building is only partial used for PoW it does not need to be rendered important. 
If the whole building is a PoW the tag should be at the building.
Are there examples where a surrounding building should get highlighted, too?

Perhaps we want a whitelist of building values: cathedral, church, mosque, temple, synagogue, shrine.
A building that was build as a church (but is not active used for that anymore) does not need a PoW tag, but deserves a highlight rendering.
That would need a lot of retagging for many PoW buildings... (even in the good mapped Aachen, Germany we have >95% with ["building"="yes"]["amenity"="place_of_worship"]

**Comment by math1985** [comment]
Deciding what is and what is not building=church is tricky though. Is [this](https://maps.google.com/maps?q=Betel,+Sherlock+Street,+Birmingham,+United+Kingdom&hl=en&ll=52.471214,-1.894219&spn=0.007241,0.021136&sll=40.044438,-68.90625&sspn=37.058943,86.572266&oq=bethel,+birmingham&hq=Betel,&hnear=Sherlock+St,+Birmingham,+United+Kingdom&t=m&z=16&layer=c&cbll=52.471318,-1.894136&panoid=CV0SvggR-Cp-OTgJHHEpOg&cbp=12,134.95,,0,0.91) a building=church?

**Comment by HolgerJeromin** [comment]
building:level=1 and roof:shape=flat is not important enough for highlighting :-P

**Comment by mkoniecz** [comment]
> it didn't work well with churches that weren't P.O.W.s

This may be handled by checking also value of building tag. I was not proposing/implementing it as I have no idea whatever deconsecrated church building are still considered important buildings.

> churches as nodes inside buildings

In my experience it happens only with unoptimal tagging and with churches of marginal religions using only small part of buildings - and such PoW are not landmarks or widely known, therefore lack of special rendering is OK.

> Deciding what is and what is not building=church is tricky though. Is this a building=church?

For me it is clearly not a building=church. Nothing indicated that this building was specially constructed as a church, PoW is not even using entire building.

Obviously, religion is a complicated topic and things drastically vary between religions and regions - but currently I think that considering building with PoW as important was a good idea. And my intention is to render also other important buildings with this special style.

**Comment by althio** [comment]
> Deciding what is and what is not building=church is tricky though. Is this
<https://maps.google.com/maps?q=Betel,+Sherlock+Street,+Birmingham,+United+Kingdom&hl=en&ll=52.471214,-1.894219&spn=0.007241,0.021136&sll=40.044438,-68.90625&sspn=37.058943,86.572266&oq=bethel,+birmingham&hq=Betel,&hnear=Sherlock+St,+Birmingham,+United+Kingdom&t=m&z=16&layer=c&cbll=52.471318,-1.894136&panoid=CV0SvggR-Cp-OTgJHHEpOg&cbp=12,134.95,,0,0.91>
a
building=church?

Not a building=church.
And certainly not a landmark.

I agree with your tags.

https://www.openstreetmap.org/way/285157655
building=yes

https://www.openstreetmap.org/node/2888318171
amenity=place_of_worship
name=Bettel City Centre Church

**Comment by polarbearing** [comment]
Attempted to clarify PoW on buildings, non-buildings and use-changed former PoW-buildings in the wiki, on a level that appears to be consensual. 
https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dplace_of_worship

**Comment by althio** [comment]
@polarbearing 
> Attempted to clarify PoW on buildings, non-buildings and use-changed former PoW-buildings in the wiki

From what I have read: :+1:

@mkoniecz 
> I think that considering building with PoW as important was a good idea.

To be very specific:
I think that landmarks are important, so look for tag building=mosque/cathedral/church/... to determine major buildings.
I think that current use is less important for the main carto, so amenity=place_of_worship + religion=* should appear as icon.

**Comment by polarbearing** [comment]
My vision, outside of buildings, would be a colour for ceremonial PoW areas (at least those of a minimal size) as johnw had described in the mailing list, leaving room for embedding in an appropriate landuse. If the PoW colour is hamonised with the PoW-building colour that would be good for recognition. 

Consequently, a no-longer-PoW building that is still a landmark could share a colour with other landmark buildings such as castles etc.

**Comment by Rovastar** [comment]
I think amenity = place_of_worship
and
building = church, cathedral, etc
are both important and need rendering differently to other stuff. And different to each other.
However I am not sure what to do if they are both?

**Comment by Rovastar** [comment]
The problem is also one of consistent tagging.
e.g. Some have the grounds around a church as the place of worship some just have the church. 
Osm-cartos job is also to help get the best and most consistent tagging where inconsistent tagging appears in the community (often most frustratingly on a country to country basis)
I think it is best to decide here the best way of doing this and this will therefore encourage others to tag that way.

**Comment by kocio-pl** [comment]
@Rovastar: That's exactly what I wanted to say! I like the example schema for school area:

http://wiki.openstreetmap.org/wiki/File:Amenity_school_usage_example.svg

It makes things very easy to understand and tag accordingly. I like to have something like that for religious areas. The amenity=place_of_worship wiki ( https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dplace_of_worship ) is a nice guide, but few days ago I finally understood how it relates to landuse=religious:

https://lists.openstreetmap.org/pipermail/tagging/2015-January/020845.html

So in general we have landuse=religious surrounding (which is not rendered now at all - though it can be quite large), inside there can be some amenity=place_of_worship field (which is rendered with some icons), some building/parkings etc., and a main building also being PoW, which is specially rendered, also with icons. That means we have two suboptimal renderings: sometimes no visible area (landuse=religious) and sometimes two icons, possibly eclipsing each other or just crowded (amenity=place_of_worship and amenity=place_of_worship+building=* - like here: http://www.openstreetmap.org/way/35014268). 

School case is easier, of course, because there's only one type of area and no icons involved, but I'd like to know at least how to avoid the most common unpleasant rendering effects.

**Comment by althio** [comment]
> The problem is also one of consistent tagging.
+
> Osm-cartos job is also to help get the best and most consistent tagging where inconsistent tagging appears in the community
+
> choose the best way of doing this and this will therefore encourage others to tag that way.

Yes

> I think it is best to decide here the best way of doing this

Is it here the best place? I don't think so. I think it is best to follow guidances from wiki or discuss on wiki/tagging to choose the best way of doing this.

**Comment by Rovastar** [comment]
I should have said using best practices. 
I have always been a fan of that.
Looking how editors do things, taginfo, wiki, etc all help us.
However opening up to the whole world for endless discussion where 1 veto with mean no change at all (often the tagging list way) is not how things are generally done here at all. If they were nothing would have been done.

Is it here the best place to decide what goes osm-carto? 
Sure it is. ;-)

**Comment by pnorman** [comment]
> @Rovastar: That's exactly what I wanted to say! I like the example schema for school area:
> 
> http://wiki.openstreetmap.org/wiki/File:Amenity_school_usage_example.svg

Yes - unfortunately, P.O.W. tagging is nowhere near that settled, and we have to make to do with what we have.

**Comment by kocio-pl** [comment]
@pnorman: Sure, and - as I mentioned - it's also more complicated scenario and includes icons, so maybe nobody knows for now, but any hint will be helpful. 

Hm - maybe it would be enough to start rendering landuse=religious lighter than pow field and with no icon, because (for example) this & pow building are the most common occurence (so we see the field, icons don't clash, and if we even have a pow field, it still will be visible)? If that would work most of the time, I would not care if there will be always some corner cases, because now it's worse.

**Comment by polarbearing** [comment]
> School case is easier, of course

Yes it is easier because it is a different concept. A school is the full institution on the ground that you can build a fence around, including car park and the janitor's shed. If somebody would propose a Place_of_Education, that would be just the classroom and the physical education pitch -- just for comparison with PoW.

PoW is the ceremonial place where people congregate and/or pray, indoors or outdoors. It cannot comprise parking and sheds. That's what we need landuse=religion for.

**Comment by kocio-pl** [comment]
I made a quick and dirty POW visual example, based on the Tom Pfeifer mail (general scheme) and school visualization (source image):

http://wiki.openstreetmap.org/wiki/File:Amenity_pow_usage_example.svg

The colours are just rough estimation and the icons are not visible here, but it resolves my 2 problems - and more:
- landuse is visible (only as a background color), but does not mix with the ceremonial area inside
- icons do not clash (they are only on objects with religion=* tags, and I guess most of the time they are separate)
- we can have multiple religion objects inside the common landuse (IIRC, someone on the list gave an example of Japan, where the temples of two different religions share the same general space)

What you think about it? Any feedback will be welcome.

**Comment by polarbearing** [comment]
Examples collected on the talk page https://wiki.openstreetmap.org/wiki/Talk:Tag:landuse%3Dreligious including your drawing, thanks.

**Comment by polarbearing** [comment]
@kocio-pl made an important point: the religion=* tags should only be on the PoW items, not on the landuse. At least when there is a PoW contained. I'd second that idea. 

Leaves only the potential nesting of PoW (consecrated area) around an active church as in the Italian case. @dieterdreist -- are there cases of an inactive church on an active consecrated area?

**Comment by davidgumberg** [comment]
@polarbearing Why is that? Are you staying safe in the possibility that you have a multireligion landuse? The land is being used for that particular religion imo, and so the tag should be there.

**Comment by kocio-pl** [comment]
Feel free to improve the image, of course, as the discussion evolves. I did it to make some things clear for me and help us find the issues we would face.

@davidgumberg: You're probably right, landuse should be tagged with religion=* tag, if there's only one religion use inside. We just should not have the icon there to avoid clash (I try to consider rendering issues).

@polarbearing: Nesting of PoW is a corner case for me - I don't have the idea how to avoid icon clash in such cases. I guess some helper, render-specific tag would be needed for that (like render:icon=no). 

And maybe that helper tag would be the common toolbox of rendering PoW areas, though I'd like to avoid it, since it's not the elegant and generic way. Probably the best thing would be fully automatic resolving rendering issues at the renderer level (like maybe "don't render this icon if surrounding PoW has the same icon" rule), but I don't know if its easy to do.

**Comment by mkoniecz** [comment]
@kocio-pl render-specific tags are by definition tagging for renderer - and therefore a bad idea.

**Comment by kocio-pl** [comment]
I know and I also don't like it too much, that's why I look for the other way. However the Eiffel Tower was hacked like this after fatal rendering decision that the walkable space should be always on top, irrespective of the layering (part of the pedestrian area is tagged render=no). So sometimes even nasty hacks are used in real life and I would not get rid of them easily, since the non-hacked rendering is even nastier...

**Comment by mkoniecz** [comment]
@kocio-pl I know about case of valid nesting PoW (I even mapped it), and in that case rendering religion symbol twice is weird looking - but it is a correct way of displaying strange situation.

![selection_002](https://cloud.githubusercontent.com/assets/899988/5691335/7435a778-98c1-11e4-9746-b3b6e67a25bf.png)

**Comment by kocio-pl** [comment]
That is visually correct for me (even if a bit strange), but the centers of both PoW are not too close here. The question is if that's also the common case in Italy (as @dieterdreist mentioned), or are they typically more symmetrical? If the first one, I wouldn't be worried and I would only care for rendering landuse=religious area in a shade of gray.

**Comment by mkoniecz** [comment]
> in Italy you can find a lot of the "opposite": an area for the place_of_worship (consecrated area) with a smaller area building=church for the actual church building inside it.

Why it is done? Is it because religious rituals are frequently not limited to church building? Or is it some kind of widespread tagging mistake?

BTW, 79% PoW in Italy is tagged on buildings (259k out of 328k - http://overpass-turbo.eu/s/6Xi, http://overpass-turbo.eu/s/6Xj).

To compare: 8% of PoW in Poland is without building tag and nearly all of them are tagging mistakes.

**Comment by dieterdreist** [comment]
2015-01-10 3:11 GMT+01:00 polarbearing <notifications@github.com>:

> Leaves only the potential nesting of PoW (consecrated area) around an
> active church as in the Italian case. @dieterdreist
> <https://github.com/dieterdreist> -- are there cases of an inactive
> church on an active consecrated area?
>


I don't think this is possible. Either a church is itself a / or within a
bigger consecrated area (and hence it is to be considered "active") or it
isn't, i.e. it has been desecrated deliberately.

**Comment by kocio-pl** [comment]
If I understand correctly, that means that if we have nested place_of_worship, we should render only one icon (I guess the inner one) to avoid possible clash.

**Comment by mkoniecz** [comment]
> are there cases of an inactive church on an active consecrated area

It may happen with old/temporary church replaced by a new bigger one. Old one is no longer used but still within something that I would tag as landuse=religious.

**Comment by kocio-pl** [comment]
That case is easy - it is already on my schema image (the other building on the landuse=religious). The problem is only nested pow area/building beacause of the icons.

**Comment by polarbearing** [comment]
Maybe we should simply distinguish three cases in relation of a building to a PoW:
1. PoW inside a building, as a node or smaller area
2. PoW identical with the building
3. PoW larger than the building

Thus for case 3, the building would be considered in use for the religion if not tagged otherwise (theatre, ruins, abandoned). This avoids nesting of PoW, and be consistent with other amenities that should not have a node of the same inside an area.

**Comment by dieterdreist** [comment]
> Am 14.01.2015 um 07:40 schrieb Mateusz Konieczny <notifications@github.com>:
> 
> It may happen with old/temporary church replaced by a new bigger one. Old one is no longer used but still within something that I would tag as landuse=religious.
> 


do you know of any real world example for this idea? Has the church in this case been desecrated? =

**Comment by mkoniecz** [comment]
> do you know of any real world example for this idea? Has the church in this case been desecrated?

Yes, I am not sure about details but old church was for some time used as warehouse, then it was moved/demolished.

**Comment by HolgerJeromin** [comment]
Proposal for rendering 
Pow way with building =yes could be rendered as normal building but with the icon. 

Only building = church, Cathedral, mosque,..  get the prominent darker fill. 

Pow without building stays like today. 

Building = church with no pow gets no darker fill as the pow could be a separate node or it is no pow anymore. 

That would promote correct building tag and somewhat hide the small pow which are no landmarks. 

We have to decide if building = Chapel is considered same as building =yes or building = church.

**Comment by dieterdreist** [comment]
sent from a phone

> Am 09.10.2015 um 01:19 schrieb Holger Jeromin <notifications@github.com>:
> 
> We have to decide if building = Chapel is considered same as building =yes or building = church.


chapel like church, size is implicit

**Comment by wmyrda** [comment]
As thread seems to include talk about landuse=religious as well I would like to take this opportunity to ask whether there is a chance to include rendering of that area in the city? Lack of it usually looks like drawing mistake or map incompleteness especially in otherwise somewhat well drawn areas like http://www.openstreetmap.org/#map=17/49.46196/22.40529 and looks very bad in the city with many historic/religious objects like http://www.openstreetmap.org/#map=17/49.77938/22.76969

I would welcome change that will render it as there is about 9000 religious areas drawn worldwide which is twice as much than for example leisure=dog_park, yet the latter one is rendered while the first one being way more important is not.

**Comment by kocio-pl** [comment]
The main problem is how should it be rendered?

**Comment by wmyrda** [comment]
If there is no other idea than at least render it as other amenities or landuse=residential. If I would be to propose idea for different color than using same color as place of worship would be st option. It is grey like landuse, but it is darker.
Good side of one of those three options is that we are not introducing new color to the map, but still we gain other area as rendered.

**Comment by kocio-pl** [comment]
Coding it will be an easy task, the most important is discussion and deciding:
1. residential color
2. place of worship color

What is the third option you mentioned? Maybe so called [societal color](https://github.com/gravitystorm/openstreetmap-carto/issues/2087#issuecomment-195417555)?

**Comment by wmyrda** [comment]
third option looks like following http://www.openstreetmap.org/#map=19/49.78595/22.76759 Seems fine for me as color is distinctive and used already in the map

**Comment by kocio-pl** [comment]
This is exactly 2. (when p.o.w. is not a building).

I think you should open new ticket, since this one has nothing to do with `landuse=religious`.

**Comment by wmyrda** [comment]
That second option was third one mentioned by me ;) Other idea was to use color just like for amenities (schools, universities, stadiums etc.)

**Comment by polarbearing** [comment]
My pref would be gray for landuse=religious, slightly darker than landuse=residential. Would that collide with anything?

**Comment by ibnteshfin** [comment]
Is there any update on landuse=religious rendering ? It almost has 10K occurrences and still is not displayed. Why not just render it like places of worships that are not building, as proposed above. It would at least prevent people from tagging for the renderer and having two icons displayed.

**Comment by polarbearing** [comment]
No, landuse=religious, PoW(no building) and PoW(building) should be different, because each form of PoW can be on a larger campus tagged landuse.

**Comment by dieterdreist** [comment]
2017-03-20 23:16 GMT+01:00 ibnteshfin <notifications@github.com>:

> Is there any update on landuse=religious rendering ? It almost has 10K
> occurrences and still is not displayed.



I still don't understand when this tag has to be applied. What are
"religious purposes"? Is this anything that has to do with the church? Is
the chapel in a train station a religious purpose? Is a hotel set up for
pilgrimage a religious purpose? Is a place where the local comunity meets a
religious purpose? Should St. Peter's Square in Rome get the
landuse=religious tag (there are often services, but besides this it is
just a square, i.e. landuse=highway). Should a catholic run kindergarten
get the tag? IMHO "religious" is not a landuse, it is a property, and we
have the religion=* tag to express it, it is used 839.000 times.

**Comment by jeisenbe** [comment]
The original point of this issue was to discuss if a building with a place_of_worship node inside should be rendered differently. As @pnorman said:

> I can write code that performs the spatial queries for checking for P.O.W. POIs inside a building, but we need to figure out if that's what we want.

Are we interested in that still? 

Currently the special "major building" rendering for place_of_worship is only done if the same polygon is tagged as `building=*` and `amenity=place_of_worship`

If that is sufficient, then this issue can be closed.

**Comment by dieterdreist** [comment]
sent from a phone

> On 9. Nov 2019, at 07:19, Joseph E <notifications@github.com> wrote:
> 
> If that is sufficient, then this issue can be closed.


IMHO a pow inside a building is not sufficient, while a building=church/cathedral/mosque etc. would be even without the amenity node.
